modules/nf-core/magus/align/main.nf

```diff
@@ -0,0 +1,58 @@
+process MAGUS_ALIGN {
+    tag "$meta.id"
+    label 'process_high'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mulled-v2-ae4ea1182e75371808710b6c081bef8b228c4815:f77664a2cdf067ddff5dc016cfa26dc129fe8c70-0':
+        'biocontainers/mulled-v2-ae4ea1182e75371808710b6c081bef8b228c4815:f77664a2cdf067ddff5dc016cfa26dc129fe8c70-0' }"
+
+    input:
+    tuple val(meta) , path(fasta)
+    tuple val(meta2), path(tree)
+    val(compress)
+
+    output:
+    tuple val(meta), path("*.aln{.gz,}"), emit: alignment
+    path "versions.yml"                 , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def loadtree = tree ? "-t $tree" : ''
+    def write_output = compress ? "--overwrite -o >(pigz -cp ${task.cpus} > ${prefix}.aln.gz)" : "-o ${prefix}.aln"
```

Comment:
I agree, that would be the ideal way to do it – this solution is a compromise, because many downstream tools don't support compressed files.
There was a fairly long thread discussing this in the slack:
https://nfcore.slack.com/archives/CJRH30T6V/p1706010248794839

Reply:
Allright, fine by me :p